frontend/src/pages/Admin/TrainingReports.js

```diff
@@ -0,0 +1,181 @@
+    'Target Population(s)',
+    'Audience',
+    'Overall Vision/Goal for the PD Event',
+  ];
+
+  const requiredCsvHeaders = [
+    'Event ID',
+    'Edit Title',
+    'Creator',
+  ];
+
+  const importTr = async () => {
+    try {
+      // Reset summary info.
+      setSkipped([]);
+      setErrors([]);
+
+      // Get the file from the file input from ref.
+      const { files } = fileInputRef.current;
+      const file = files[0];
+
+      // Get File.
+      const data = new FormData();
+      data.append('file', file);
+      const res = await importTrainingReports(file);
+      setSuccess(`${res.count} events imported successfully.`);
+      setSkipped(res.skipped);
+      setErrors(res.errors);
+      setError('');
+    } catch (err) {
+      setError('Error attempting to import training reports.');
+    } finally {
+      // Clear file input.
+      setInfo('');
+      setUploadDisabled(true);
+    }
+  };
+
+  const onChange = (e) => {
+    const { files } = e.target;
+
+    // Clear error and success messages.
+    setError('');
+    setSuccess('');
+    setInfo('');
+
+    // Verify that the file is a CSV.
+    if (files.length === 1 && files[0].type !== 'text/csv') {
+      setError('Please upload a CSV file.');
+      setUploadDisabled(true);
+      e.target.value = null;
+      return;
+    }
+
+    const reader = new FileReader();
+    reader.onload = (event) => {
+      // Load the csv.
+      const csv = event.target.result;
+
+      // Replace all instances of LF that are not followed by a CR with a space.
+      const cleanCsv = csv.replace(/\r(?!\n)/g, ' ');
+
+      // Get all headers.
+      const headers = cleanCsv.split('\r\n')[0].split(',');
+
+      // Validate columns.
+      const requiredHeaders = requiredCsvHeaders.filter((header) => !headers.includes(header));
+      const invalidHeaders = headers.filter((header) => !validCsvHeaders.includes(header));
+      if (requiredHeaders.length > 0) {
+        setError(`Required headers missing: ${requiredHeaders.join(', ')}`);
+        setUploadDisabled(true);
+        e.target.value = null;
+        return;
+      }
+      if (invalidHeaders.length > 0) {
+        setError(`Invalid headers found: ${invalidHeaders.join(', ')}`);
+        setUploadDisabled(true);
+        e.target.value = null;
+        return;
+      }
+
+      // Get a distinct count of 'Event ID' column's.
+      const eventIds = cleanCsv.split('\r\n').map((row) => row.split(',')[1]).filter((id) => id !== 'Event ID' && id !== undefined);
+      const distinctEventIds = [...new Set(eventIds)];
+
+      // Check if we have duplicate 'Event ID' values.
+      if (distinctEventIds.length !== eventIds.length) {
+        setError('Duplicate Event IDs found. Please correct and try again.');
+        setUploadDisabled(true);
+        e.target.value = null;
+        return;
+      }
+
+      // Display count of distinct ids.
+      setInfo(`${distinctEventIds.length} events will be imported.`);
```

Comment:
If the file only has headers and no rows, you'll see this:

![Screenshot 2023-09-07 at 3 11 53 PM](https://github.com/HHS/Head-Start-TTADP/assets/17074357/522c5da4-e283-46ba-88ec-4969c70a5c12)

Not really a big deal or bug IMO, but thought I'd mention it.

Reply:
Thought about it - not really worth commenting on this.

Reply:
I can add a check.

Reply:
Per discussion will leave as is for now.